### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'templated_str'

### ISSUE THREAD

**Issue by nolanbconaway** [sqlfluff/sqlfluff#1196]
A bug surfaced in slack: I have no idea what's going on here! The SQL is

```sql
--Airflow: ml_cxe_service_metrics -
CREATE TABLE IF NOT EXISTS {{ params.ml_schema }}.daily_service_metrics (
    source_type varchar(10) NOT NULL,
    visit_date date NOT NULL,
    event_name text NOT NULL,
    event_count bigint NOT NULL
)
diststyle even
compound sortkey(visit_date, source_type)
;
```

## Expected Behaviour

No errors!

## Observed Behaviour

linting:

```
➜ echo '--Airflow: ml_cxe_service_metrics -
CREATE TABLE IF NOT EXISTS {{ params.ml_schema }}.daily_service_metrics (
    source_type varchar(10) NOT NULL,
    visit_date date NOT NULL,
    event_name text NOT NULL,
    event_count bigint NOT NULL
)
diststyle even
compound sortkey(visit_date, source_type)
;' | sqlfluff lint - --verbose
==== sqlfluff ====
sqlfluff:                0.6.0 python:                  3.8.7
implementation:        cpython dialect:                  ansi
verbosity:                   1 templater:               jinja

== [stdin] FAIL
L:   0 | P:   0 |  TMP | Unrecoverable failure in Jinja templating: 'params' is
                       | undefined. Have you configured your variables?
                       | https://docs.sqlfluff.com/en/latest/configuration.html
L:   2 | P:  31 |  TMP | Undefined jinja template variable: 'params'
```

fixing:
```
➜ echo '--Airflow: ml_cxe_service_metrics -
CREATE TABLE IF NOT EXISTS {{ params.ml_schema }}.daily_service_metrics (
    source_type varchar(10) NOT NULL,
    visit_date date NOT NULL,
    event_name text NOT NULL,
    event_count bigint NOT NULL
)
diststyle even
compound sortkey(visit_date, source_type)
;' | sqlfluff fix - --dialect postgres
Traceback (most recent call last):
...
AttributeError: 'NoneType' object has no attribute 'templated_str
```

`params` is not defined in my (empty) project, but the fixer should not except out!

## Steps to Reproduce

see above!

## Version

0.6.0

## Configuration

zilch

**Comment by barrywhart** [comment]
I can't reproduce this using the instructions above, even after doing `git checkout 0.6.0`. Can you try it again, @nolanbconaway?

**Comment by nolanbconaway** [comment]
Yeah- I can still reproduce!

```

➜ echo '--Airflow: ml_cxe_service_metrics -
CREATE TABLE IF NOT EXISTS {{ params.ml_schema }}.daily_service_metrics (
    source_type varchar(10) NOT NULL,
    visit_date date NOT NULL,
    event_name text NOT NULL,
    event_count bigint NOT NULL
)
diststyle even
compound sortkey(visit_date, source_type)
;' | sqlfluff fix - --verbose
Traceback (most recent call last):
  File "/home/squidward/.pyenv/versions/3.8.7/bin/sqlfluff", line 8, in <module>
    sys.exit(cli())
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlfluff/cli/commands.py", line 444, in fix
    stdout = result.paths[0].files[0].fix_string()[0]
  File "/home/squidward/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlfluff/core/linter/linted_file.py", line 241, in fix_string
    linter_logger.debug("Original Tree: %r", self.templated_file.templated_str)
AttributeError: 'NoneType' object has no attribute 'templated_str'

➜ sqlfluff --version
sqlfluff, version 0.6.0


```

**Comment by barrywhart** [comment]
That fails for me, too. I notice that the original example in the issue description uses `sqlfluff lint`, while just now you did `sqlfluff fix`.

 In this case, there's no file to update in place, which is the normal behavior of `fix`. I suspect that's why it's failing. What behavior would you expect in this case? Should it refuse to fix SQL coming from stdin? Or should it print the fixed SQL to  stdout?

To confirm this is what's happening, I put the SQL example code in a file `test.sql` and ran `sqlfluff fix test.sql`. This runs with no error.

**Comment by nolanbconaway** [comment]
Interesting! It does not except out via file but only via stdin? We set up stdin to print the fixed sql to stdout so we still shouldn't be excepting. In any case I've been shepherding the stdin/out cli so I can handle this one
